packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1426,6 +1474,85 @@ export class TransactionController extends BaseController<
 
     return dappSuggestedGasFees;
   }
+
+  /**
+   * Validates and adds external provided transaction to state.
+   *
+   * @param transactionMeta - Nominated external transaction to be added to state.
+   */
+  private async addExternalTransaction(transactionMeta: TransactionMeta) {
+    const { networkId, chainId } = this.getChainAndNetworkId();
+    const fromAddress = transactionMeta?.transaction?.from;
+    const confirmedTxs = this.state.transactions.filter(
+      (transaction) =>
+        transaction.transaction.from === fromAddress &&
```

Comment:
Is it worth avoiding some duplication and first creating a `sameFromAndNetworkTransactions` so `confirmedTxs` and `pendingTxs` only have to filter that based on status?